lib/src/pooling/pool_object.dart

```diff
@@ -1,11 +1,15 @@
 part of oxygen;
 
 abstract class PoolObject<T> {
+  /// The pool from which the object came from.
   ObjectPool _pool;
 
+  /// Initialize this object.
   void init([T data]);
```

Comment:
is data the first element of the pool?

Reply:
No `data` is the data you pass to `init` to initialise the `PoolObject`. Whenever a `ObjectPool` acquires a new `PoolObject` :-)